### PR TITLE
Fix a whole bunch of warnings

### PR DESCRIPTION
**Pull request by YaLTeR**

### CODE REVIEW

File: source/application/DbgHelp/DbgHelpTypes.h [hunk 1]
```diff
@@ -73,6 +74,12 @@ class DbgHelpBasicType
             return L"float"s;
         case BasicType::Double:
             return L"double"s;
+        default:
+            /*
+             * VS is stupid and warns even if everything is covered.
```
Comment (Warepire) [inline source/application/DbgHelp/DbgHelpTypes.h]: Can we drop these comments?

File: source/application/DbgHelp/DbgHelpTypes.h [hunk 2]
```diff
@@ -10,7 +10,14 @@
 #include <string>
 #include <variant>
 
+/*
+ * Make the pointless warning shut up. This can be removed when the bug below is fixed.
```
Comment (Warepire) [inline source/application/DbgHelp/DbgHelpTypes.h]: Can we reword this to something like "This triggers the warning erroneously due to a bug: -link-"?

**Review by Warepire — CHANGES_REQUESTED** [review]

File: source/application/DbgHelp/DbgHelpTypes.h (hunk 1 above)
Comment (YaLTeR) [reply source/application/DbgHelp/DbgHelpTypes.h]: Sure.

File: source/application/DbgHelp/DbgHelpTypes.h (hunk 2 above)
Comment (YaLTeR) [reply source/application/DbgHelp/DbgHelpTypes.h]: Yeah, I'll do that.

**Comment by Warepire** [comment]
Looks good, please squash the changes into the original commits and then I will merge.

**Comment by YaLTeR** [comment]
Should be done, hopefully I didn't mess anything up.

**Review by Warepire — APPROVED** [review]
